### ISSUE TITLE
Hot Module Replacement does not work with only one file

### ISSUE THREAD

**Issue by SerhiiBilyk** [webpack/webpack-dev-server#1535]
* Operating System: Windows 10
* Node Version: 10.10.0
* NPM Version:6.4.0
* webpack Version:4.21.0
* webpack-dev-server Version:3.1.5

- [x] This is a **bug**
- [] This is a **modification** request

### Code

```js

const APP_DIR = path.resolve(__dirname, "./packages");
const BUILD_DIR = path.resolve(__dirname, "./docs");

module.exports = env => {
  console.log('env.mode', env.mode)
  return smp.wrap({
    entry: {
      app: [/*"babel-polyfill", */"./packages/index.js"],
      vendors: "./packages/vendors.cache.js",
      css: "./packages/vendors.css.js",
    },

    devtool: "eval-source-map",
    output: {
      path: BUILD_DIR,
      filename: "[name].[hash].js",
    },
    devServer: {
      hot: true,
      contentBase: APP_DIR, 
      port: 8081
    },
    node: {
      fs: "empty"
    },
    stats: 'none',
    optimization: {
      splitChunks: {
        chunks: (chunk) => chunk.name !== 'css'
      },
      minimize: true
    },
    module: {
      rules: [{
          test: /\.scss$/,
          use: [
            MiniCssExtractPlugin.loader,
            {
              loader: 'css-loader',
              options: {
                modules: true,
                sourceMap: true,
                importLoader: 3,
                localIdentName: '[local]___[hash:base64:5]'
              }
            },
            'sass-loader',
            {
              loader: 'sass-resources-loader',
              options: {
                resources: `${APP_DIR}/css/sass/mixins.scss`
              }
            }
          ]
        },
        {
          test: /\.(js|jsx)$/,
          include: APP_DIR,
          use: {
            loader: 'babel-loader',
            options: {
              plugins: ['syntax-dynamic-import']
            }
          }
        },
        {
          test: /\.(cache.js)$/,
          use: [
            'happypack/loader?id=cache'
          ],
        },
        {
          test: /\.css$/,
          use: [{
            loader: MiniCssExtractPlugin.loader,
            options: {
              publicPath: '../'
            }
          }, 'happypack/loader?id=css']

        },
        {
          test: /\.(jpg|png|svg)$/,
          use: 'happypack/loader?id=file'
        },
        {
          test: /\.(eot|woff|woff2|ttf|gif)(\?\S*)?$/,
          loader: "url-loader?limit=100000"
        },
        {
          test: /\.(html)$/,
          use: {
            loader: "html-loader"
          }
        },
        {
          test: require.resolve("angular"),
          loader: "exports-loader?window.angular"
        },
        {
          test: require.resolve("./packages/js/cic/config/app.js"),
          use: "exports-loader?app"
        },
        {
          test: require.resolve("jquery"),
          use: [{
              loader: "expose-loader",
              options: "jQuery"
            },
            {
              loader: "expose-loader",
              options: "$"
            },
            {
              loader: "expose-loader",
              options: "jquery"
            }
          ]
        },

      ]
    },
    resolve: {
      alias: {
        api: path.resolve(__dirname, `${APP_DIR}/js/api`),
        utils: path.resolve(__dirname, `${APP_DIR}/js/cic/utils`),
        actions: path.resolve(__dirname, `${APP_DIR}/js/cic/actions`),
        directives: path.resolve(__dirname, `${APP_DIR}/js/cic/directives`),
      },
      extensions: [".js", ".html", ".css", ".png", ".svg", ".appcache"]
    },
    plugins: [

      new HtmlWebpackPlugin({
        title: "Development mode",
        template: "assets/dev.index.ejs",  
        chunksSortMode: 'none',
        files: {
          jquery: {
            entry: `jquery.dll.js`
          },
        }
      }),
      new webpack.ProvidePlugin({
        angular: "angular",
        jquery: "jquery",
        $: "jquery",
        jQuery: "jquery",
        "window.jQuery": "jquery",
        moment: 'moment'
      }),
      new MiniCssExtractPlugin({
        filename: "[name].css",
        chunkFilename: "[id].css",
      }),
      new HappyPack({
        id: 'css',
        threadPool: happyThreadPool,

        loaders: ['css-loader']
      }),
      new HappyPack({
        id: 'cache',
        threadPool: happyThreadPool,
        loaders: ['cache-loader']
      }),
      new HappyPack({
        id: 'file',
        threadPool: happyThreadPool,
        loaders: ['file-loader']
      }),
      new webpack.DefinePlugin({
        'process.env': {
          NODE_ENV: JSON.stringify(process.env.NODE_ENV)
        }
      }),
      new webpack.DefinePlugin({
        '__NODE_ENV__': {
          mode: JSON.stringify(env.mode),
        }
      }),
      new CopyWebpackPlugin([{
          from: `${APP_DIR}/img`,
          to: `${BUILD_DIR}/img`,
          toType: 'dir'
        },
        {
          from: `${APP_DIR}/partials`,
          to: `${BUILD_DIR}/partials`,
          toType: 'dir'
        },
        {
          from: `${APP_DIR}/js/cic`,
          to: `${BUILD_DIR}/js/cic`,
          ignore: ['vendor/**/*'],
          toType: 'dir'
        },
        {
          from: `${APP_DIR}/js/surveys`,
          to: `${BUILD_DIR}/js/surveys`,
          toType: 'dir'
        },
        {
          from: `${APP_DIR}/js/templateManager`,
          to: `${BUILD_DIR}/js/templateManager`,
          toType: 'dir'
        },
        {
          from: `${APP_DIR}/js/tasker`,
          to: `${BUILD_DIR}/js/tasker`,
          toType: 'dir'
        },
        {
          from: `${APP_DIR}/js/mail_channel`,
          to: `${BUILD_DIR}/js/mail_channel`,
          toType: 'dir'
        }
      ])
    ]
  } );
}
```

### Expected Behavior 
Rebuild on every change of file content 

### Actual Behavior

Very strange behavior. HMR does not update only one file [./packages/js/cic/lazyLoad.js] in project. Rest file are updating successfully.
[WDS] App updated. Recompiling...
[WDS] App hot update...
[HMR] Checking for updates on the server...
[HMR] Nothing hot updated.
[HMR] App is up to date.
[HMR] Nothing hot updated.
[HMR] App is up to date.
[HMR] Nothing hot updated.
[HMR] App is up to date.

lazyLoad.js
```js
const rootReducer = require('./reducers/api.js');
const communicatorReducer=require('./reducers/communicator.js')
const thunk = require('redux-thunk').default;
const ngRedux = require('ng-redux');

app.config(($ngReduxProvider, $ocLazyLoadProvider, $stateProvider) => {


    $ocLazyLoadProvider.config({
   
        serie: true,
    
    })

    $stateProvider
        .state('tout', {
            url: "/",
            templateUrl: 'partials/cic/tout/index.html',
            resolve: {
                service: ['$ocLazyLoad', ($ocLazyLoad) =>
                    import('../cic/services/toutService.js')
                    .then(mod => $ocLazyLoad.load(mod.default))
                ],
                ctrl: ['$ocLazyLoad', ($ocLazyLoad) =>
                    import('../cic/controllers/tout.js')
                    .then(mod => $ocLazyLoad.load(mod.default))
                ],
            },
            controller: 'ToutCtrl'
        })
 
    $ngReduxProvider.createStoreWith({
        api: rootReducer,
        communicator:communicatorReducer
    }, [thunk]);

})
```

Sorry, but I can't share the whole application.

**Comment by evilebottnawi** [comment]
@SerhiiBilyk Can't reproduce, need minimum reproducible test repo

**Comment by evilebottnawi** [comment]
Also `MiniCssExtractPlugin` doesn't support hmr

**Comment by SerhiiBilyk** [comment]
@evilebottnawi thank you for your fast response, I will share tomorrow a minimum reproducible repo

**Comment by SerhiiBilyk** [comment]
I have created reproducible repo, and it works fine.
When I'm starting dev-server in VSCode terminal it works as expected, unlike in windows cmd.
Now dev server in main app works fine.
I don't know how to explain it. When I will understand what's happened I will reopen issue and reproduce all my steps.
@evilebottnawi Before all steps, I have replaced MiniCssExtractPlugin to 'style-loader'.  
Now, I'm using MiniCssExtractPlugin  only in production build.
